node/stateSync.go

```diff
@@ -0,0 +1,201 @@
+package node
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/nknorg/nkn/v2/chain/db"
+	"github.com/nknorg/nkn/v2/util/log"
+
+	"github.com/nknorg/nkn/v2/config"
+
+	"github.com/nknorg/nkn/v2/chain/trie"
+	"github.com/nknorg/nkn/v2/common"
+)
+
+type stateSync struct {
+	d *Downloader // Downloader instance to access and manage current peerset
+
+	root   common.Uint256 // State root currently being synced
+	sched  *trie.Sync     // State trie sync scheduler defining the tasks
+	hasher func([]byte) []byte
+
+	done chan struct{} // Channel to signal termination completion
+	err  error         // Any error hit during sync (set before completion)
+
+	hashesCh       chan []common.Uint256
+	stateCh        chan [][]byte // Response states channel
+	timeoutReq     chan []common.Uint256
+	failedHashesCh chan []common.Uint256
+	stopChans      []chan struct{}
```

Comment:
All workers can share the same stop channel, just close it when need to send stop signal